File: atv9/exerc9.go
package main
import "fmt"

type Imprimivel interface {
    Imprimir()
}

type Relatorio struct{}

func (Relatorio) Imprimir() {
    fmt.Println("Imprimindo relatório...")
}

type Contrato struct{}

func (Contrato) Imprimir() {
    fmt.Println("Imprimindo contrato...")
}

func main() {
    var r Imprimivel = Relatorio{}
    var c Imprimivel = Contrato{}

    r.Imprimir()
    c.Imprimir()
}

File: atv9/exerc9.py
from typing import Protocol

class Imprimivel(Protocol):
    def imprimir(self) -> None:
        ...

class Relatorio:
    def imprimir(self) -> None:
        print("Imprimindo relatório...")

class Contrato:
    def imprimir(self) -> None:
        print("Imprimindo contrato...")

def imprimir_documento(doc: Imprimivel) -> None:
    doc.imprimir()

relatorio = Relatorio()
contrato = Contrato()

imprimir_documento(relatorio)
imprimir_documento(contrato)
